packages/ipfs-message-port-server/src/server.js

```diff
@@ -212,11 +213,14 @@ exports.Server = class Server {
     if (!query.signal.aborted) {
       try {
         const value = await query.result
-        const transfer = [...new Set(value.transfer || [])]
+        const transfer = ensureNonDetachedUniqueBuffers(value.transfer || [])
```

Comment:
I'll provide more input in #3480 but I do think this solves actual problem, it just pushes it further down the stack.